provider/google.go

```diff
@@ -204,18 +204,14 @@ func (p *GoogleProvider) Records() (endpoints []*endpoint.Endpoint, _ error) {
 			if !supportedRecordType(r.Type) {
 				continue
 			}
-			ep := &endpoint.Endpoint{
-				DNSName:    strings.TrimSuffix(r.Name, "."),
-				RecordType: r.Type,
-				Targets:    make(endpoint.Targets, 0, len(r.Rrdatas)),
-				RecordTTL:  endpoint.TTL(r.Ttl),
-			}
+			targets := make(endpoint.Targets, 0, len(r.Rrdatas))
+
 			for _, rr := range r.Rrdatas {
 				// each page is processed sequentially, no need for a mutex here.
-				ep.Targets = append(ep.Targets, strings.TrimSuffix(rr, "."))
+				targets = append(targets, rr)
 			}
-			sort.Sort(ep.Targets)
-			endpoints = append(endpoints, ep)
+			sort.Sort(targets)
+			endpoints = append(endpoints, endpoint.NewEndpointWithTTL(r.Name, r.Type, endpoint.TTL(r.Ttl), targets...))
```

Comment:
I think we can directly pass-in `r.Rrdatas` as the last argument here, can't we?